lib/scout_apm/layer.rb

```diff
@@ -13,13 +13,17 @@ class Layer
     #   instrumentation for an example of how this is useful
     attr_accessor :name
 
-    # An array of children layers, in call order.
+    # An array of children layers
     # For instance, if we are in a middleware, there will likely be only a single
     # child, which is another middleware.  In a Controller, we may have a handful
     # of children: [ActiveRecord, ActiveRecord, View, HTTP Call].
     #
     # This useful to get actual time spent in this layer vs. children time
-    attr_reader :children
+    #
+    # TODO: Check callers for compatibility w/ nil to avoid making an empty array
```

Comment:
It's better to just never make the empty array, since 99% of the time, a caller just calls  `.each` on it, which ends up being a noop, but I didn't want to spend the time right now to find and fix all callers.

No harm in making the array, just an operation that doesn't add to anything.  This is still a benefit because the Layer-at-rest won't hold onto an empty array.